walkeeper/src/timeline.rs

```diff
@@ -0,0 +1,262 @@
+//! This module contains tools for managing timelines.
+//!
+
+use anyhow::{bail, Result};
+use fs2::FileExt;
+use lazy_static::lazy_static;
+use log::*;
+use pageserver::ZTimelineId;
+use postgres_ffi::xlog_utils::{find_end_of_wal, TimeLineID};
+use std::cmp::{max, min};
+use std::collections::HashMap;
+use std::fs::{self, File, OpenOptions};
+use std::io::{Seek, SeekFrom};
+use std::path::Path;
+use std::sync::{Arc, Condvar, Mutex};
+use zenith_utils::bin_ser::LeSer;
+use zenith_utils::lsn::Lsn;
+
+use crate::receive_wal::{SafeKeeperInfo, CONTROL_FILE_NAME, SK_FORMAT_VERSION, SK_MAGIC};
+use crate::replication::{HotStandbyFeedback, END_REPLICATION_MARKER};
+use crate::WalAcceptorConf;
+
+/// Shared state associated with database instance (tenant)
+#[derive(Debug)]
+pub struct SharedState {
```

Comment:
Nothing in SharedState needs to be 'pub', it's not used outside the file.

Reply:
Fixed.